modules/nf-core/star/align/main.nf

```diff
@@ -21,33 +21,34 @@ process STAR_ALIGN {
-    tuple val(meta), path('*.out.junction')          , optional:true, emit: junction
-    tuple val(meta), path('*.out.sam')               , optional:true, emit: sam
-    tuple val(meta), path('*.wig')                   , optional:true, emit: wig
-    tuple val(meta), path('*.bg')                    , optional:true, emit: bedgraph
+    tuple val(meta), path('*d.out.bam')                              , optional:true, emit: bam
+    tuple val(meta), path("${prefix}.sortedByCoord.out.bam")         , optional:true, emit: bam_sorted
+    tuple val(meta), path("${prefix}.Aligned.sortedByCoord.out.bam") , optional:true, emit: bam_sorted_aligned
+    tuple val(meta), path('*toTranscriptome.out.bam')                , optional:true, emit: bam_transcript
+    tuple val(meta), path('*Aligned.unsort.out.bam')                 , optional:true, emit: bam_unsorted
+    tuple val(meta), path('*fastq.gz')                               , optional:true, emit: fastq
+    tuple val(meta), path('*.tab')                                   , optional:true, emit: tab
+    tuple val(meta), path('*.SJ.out.tab')                            , optional:true, emit: spl_junc_tab
+    tuple val(meta), path('*.ReadsPerGene.out.tab')                  , optional:true, emit: read_per_gene_tab
+    tuple val(meta), path('*.out.junction')                          , optional:true, emit: junction
+    tuple val(meta), path('*.out.sam')                               , optional:true, emit: sam
+    tuple val(meta), path('*.wig')                                   , optional:true, emit: wig
+    tuple val(meta), path('*.bg')                                    , optional:true, emit: bedgraph
 
     when:
     task.ext.when == null || task.ext.when
 
     script:
     def args = task.ext.args ?: ''
-    def prefix = task.ext.prefix ?: "${meta.id}"
+    prefix = task.ext.prefix ?: "${meta.id}"
     def reads1 = [], reads2 = []
     meta.single_end ? [reads].flatten().each{reads1 << it} : reads.eachWithIndex{ v, ix -> ( ix & 1 ? reads2 : reads1) << v }
     def ignore_gtf      = star_ignore_sjdbgtf ? '' : "--sjdbGTFfile $gtf"
     def seq_platform    = seq_platform ? "'PL:$seq_platform'" : ""
     def seq_center      = seq_center ? "'CN:$seq_center'" : ""
-    def attrRG          = args.contains("--outSAMattrRGline") ? "" : "--outSAMattrRGline 'ID:$prefix' $seq_center 'SM:$prefix' $seq_platform"
+    attrRG          = args.contains("--outSAMattrRGline") ? "" : "--outSAMattrRGline 'ID:$prefix' $seq_center 'SM:$prefix' $seq_platform"
     def out_sam_type    = (args.contains('--outSAMtype')) ? '' : '--outSAMtype BAM Unsorted'
-    def mv_unsorted_bam = (args.contains('--outSAMtype BAM Unsorted SortedByCoordinate')) ? "mv ${prefix}.Aligned.out.bam ${prefix}.Aligned.unsort.out.bam" : ''
+    mv_unsorted_bam = (args.contains('--outSAMtype BAM Unsorted SortedByCoordinate')) ? "mv ${prefix}.Aligned.out.bam ${prefix}.Aligned.unsort.out.bam" : ''
```

Comment:
Why did you also remove `def` here?

Reply:
Because prefix is now global, this line throws an error if the dependent mv_unsorted_bam is not global too

Reply:
Oh I didn't know that! Thanks :p